sdk/messaging/azservicebus/README.md

```diff
@@ -113,48 +113,79 @@ Once you've created a [Client][godoc_client] you can create a [Sender][godoc_sen
 
-```go
-sender, err := client.NewSender("<queue or topic>", nil)
-
-if err != nil {
-  panic(err)
-}
+Sending a single message:
 
-// send a single message
-err = sender.SendMessage(context.TODO(), &azservicebus.Message{
+```go
+// Send a single message
+err := sender.SendMessage(context.TODO(), &azservicebus.Message{
   Body: []byte("hello world!"),
 }, nil)
 ```
 
+#### Sending multiple messages using a batch
+
 You can also send messages in batches, which can be more efficient than sending them individually
 
 ```go
+// Sending multiple messages at a time using a MessageBatch
+
 // Create a message batch. It will automatically be sized for the Service Bus
 // Namespace's maximum message size.
-messageBatch, err := sender.NewMessageBatch(context.TODO(), nil)
+currentMessageBatch, err := sender.NewMessageBatch(context.TODO(), nil)
 
 if err != nil {
   panic(err)
 }
 
-// Add a message to our message batch. This can be called multiple times.
-err = messageBatch.AddMessage(&azservicebus.Message{
-    Body: []byte(fmt.Sprintf("hello world")),
-}, nil)
+messagesToSend := []*azservicebus.Message{
+  // any messages that you'd want to send would be here, or sourced from
+  // somewhere else.
+}
+
+for i := 0; i < len(messagesToSend); i++ {
+  // Add a message to our message batch. This can be called multiple times.
+  err = currentMessageBatch.AddMessage(messagesToSend[i], nil)
 
-if errors.Is(err, azservicebus.ErrMessageTooLarge) {
-  fmt.Printf("Message batch is full. We should send it and create a new one.\n")
+  if errors.Is(err, azservicebus.ErrMessageTooLarge) {
+    if currentMessageBatch.NumMessages() == 0 {
+      // This means the message itself is too large to be sent, even on its own.
+      // This will require intervention from the user.
+      panic("Single message is too large to be sent in a batch.")
+    }
 
-  // send what we have since the batch is full
-  err := sender.SendMessageBatch(context.TODO(), messageBatch, nil)
+    fmt.Printf("Message batch is full. Sending it and creating a new one.\n")
+
+    // send what we have since the batch is full
+    err := sender.SendMessageBatch(context.TODO(), currentMessageBatch, nil)
+
+    if err != nil {
+      panic(err)
+    }
+
+    // Create a new batch and retry adding this message to our batch.
+    newBatch, err := sender.NewMessageBatch(context.TODO(), nil)
+
+    if err != nil {
+      panic(err)
+    }
+
+    currentMessageBatch = newBatch
+
+    // rewind the counter and attempt to add the message again (this batch
+    // was full so it didn't go out with the previous SendMessageBatch call).
+    i--
+  } else if err != nil {
+    panic(err)
+  }
+}
+
+// check if any messages are remaining to be sent.
+if currentMessageBatch.NumMessages() > 0 {
+  err := sender.SendMessageBatch(context.TODO(), currentMessageBatch, nil)
```

Comment:
This is pretty big for a readme snippet. Could it be an example test instead?

Reply:
It is a bit big, but my worry is that if someone copy/pastes they'll miss the pattern they really need to follow. I think we get a lot of copy/paste programming so I just wanted to be safe.